clar2wasm/src/words/equal.rs

```diff
@@ -39,26 +39,31 @@ impl Word for IsEq {
         }
         val_locals.reverse();
 
-        // Traverse the second operand pushing it onto the stack.
-        let sec_op = args.get_expr(1)?;
-        generator.traverse_expr(builder, sec_op)?;
+        for val in &val_locals {
+            builder.local_get(*val);
+        }
+
+        if args.len() == 1 {
+            // push same first operand to stack
+            for val in &val_locals {
+                builder.local_get(*val);
+            }
```

Comment:
This seem like it will add a lot of operations in this case. For example, if we are comparing two strings, we are adding a lot of operations here.
I would have instead just traversed the first argument, check if it was the only one and if yes, push true to the stack and return.
But maybe I fail to see the big picture here...

Reply:
@Acaccia Changed for an explicit return.

You've may noticed that I am getting used to Webassembly and project details on the go. :-)

Thanks for all the guidance.

Reply:
That's nice, actually cleaner than what I had in mind. Good job!